### ISSUE TITLE
Why converting Exceptions into WP_Error?

### ISSUE THREAD

**Issue by widoz** [j-arens/wp-rest-router#5]
IIRC the `WP_Error` was a way implemented in WP time ago when Exceptions weren't a thing. 

Wouldn't be better to stay with native php way and use Exceptions since the lib seems to targeting php 7.1?

Referencing: https://github.com/j-arens/wp-rest-router/blob/master/src/WordPress/Rest/CallbackWrappers.php#L19

**Comment by j-arens** [comment]
Yeah I agree that it would be nice to not convert to `WP_Error` but WordPress does a little work behind the scenes when dispatching a rest request to detect and format `WP_Error` objects. I could be wrong but I think that letting exceptions propagate all the way up could cause WordPress's fatal error handler to kick in, which can lead to ambiguous error messages on the client at best.

This try/catch functionality is mostly just for my own convenience though. Maybe it would better to remove it and let users implement try/catch functionality themselves via a middleware or some other method?

https://github.com/WordPress/WordPress/blob/3def77672dd86568b168d5b86ee14b4acb66c42c/wp-includes/rest-api/class-wp-rest-server.php#L337
